### PR TITLE
Added Multilingual functionality

### PR DESCRIPTION
**Pull request by Adarsha-gg**
The second commit was for formatting.

### CODE REVIEW

File: AudioTranscriber.py [hunk 1]
```diff
@@ -52,14 +154,16 @@ def transcribe_audio_queue(self, audio_queue):
                 self.update_transcript(who_spoke, text, time_spoken)
                 self.transcript_changed_event.set()
 
+    def change_lang(self,language):
```
Comment (SevaSk) [inline AudioTranscriber.py:157]: space after comma

File: AudioTranscriber.py [hunk 2]
```diff
@@ -82,7 +186,11 @@ def process_speaker_data(self, data):
         return temp_file
 
     def get_transcription(self, file_path):
-        result = self.audio_model.transcribe(file_path, fp16=torch.cuda.is_available())
+        if self.lang == "en":
```
Comment (SevaSk) [inline AudioTranscriber.py:189]: We shouldn't have to call load_model every time we call get_transcription the model should only be loaded when we initialize or change the language

File: main.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import threading
-from AudioTranscriber import AudioTranscriber
+from AudioTranscriber import AudioTranscriber,LANGUAGES
```
Comment (SevaSk) [inline main.py:2]: space

File: main.py [hunk 4]
```diff
@@ -60,11 +60,14 @@ def create_ui_components(root):
     update_interval_slider.set(2)
     update_interval_slider.grid(row=3, column=1, padx=10, pady=10, sticky="nsew")
 
-    return transcript_textbox, response_textbox, update_interval_slider, update_interval_slider_label, freeze_button
+    combobox = ctk.CTkOptionMenu(root, values=list(LANGUAGES.values()))
```
Comment (SevaSk) [inline main.py:63]: Could we move the dropdown somewhere  better in the UI. We are added alot of white space here.

**Review by SevaSk — CHANGES_REQUESTED** [review]
Nice job! Left some comments for changes needed.

File: AudioTranscriber.py [hunk 5]
```diff
@@ -93,6 +93,8 @@ def update_transcript(self, who_spoke, text, time_spoken):
             transcript.insert(0, (f"{who_spoke}: [{text}]\n\n", time_spoken))
         else:
             transcript[0] = (f"{who_spoke}: [{text}]\n\n", time_spoken)
+
```
Comment (SevaSk) [inline AudioTranscriber.py:96]: blank space added?

File: TranscriberModels.py [hunk 6]
```diff
@@ -9,22 +9,34 @@ def get_model(use_api):
     else:
         return WhisperTranscriber()
 
-class WhisperTranscriber:
-    def __init__(self):
+class WhisperTranscriber():
+
```
Comment (SevaSk) [inline TranscriberModels.py:13]: blank space?

File: TranscriberModels.py [hunk 7]
```diff
@@ -9,22 +9,34 @@ def get_model(use_api):
     else:
         return WhisperTranscriber()
 
-class WhisperTranscriber:
-    def __init__(self):
+class WhisperTranscriber():
```
Comment (SevaSk) [inline TranscriberModels.py:12]: added unnecessary ()

File: TranscriberModels.py [hunk 8]
```diff
@@ -9,22 +9,34 @@ def get_model(use_api):
...
+    
+    
+    def __init__(self, language="en"):
+        self.lang = language
         self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.en.pt'))
```
Comment (SevaSk) [inline TranscriberModels.py:17]: language not always corresponding to audio_model loaded here, audio model should depend on language

File: TranscriberModels.py [hunk 9]
```diff
@@ -9,22 +9,34 @@ def get_model(use_api):
...
     def get_transcription(self, wav_file_path):
         try:
-            result = self.audio_model.transcribe(wav_file_path, fp16=torch.cuda.is_available())
+            result = self.audio_model.transcribe(wav_file_path, fp16=torch.cuda.is_available(), language = self.lang)
         except Exception as e:
             print(e)
             return ''
         return result['text'].strip()
     
+    def change_lang(self,language):
```
Comment (SevaSk) [inline TranscriberModels.py:28]: space after ,

File: TranscriberModels.py [hunk 10]
```diff
@@ -9,22 +9,34 @@ def get_model(use_api):
...
+        self.lang = language
+        if self.lang == "en":
+            self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.en.pt'))
+        else:
+            self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.pt')) 
+
 class APIWhisperTranscriber:
     def get_transcription(self, wav_file_path):
-        audio_file= open(wav_file_path, "rb")
+        new_file_path = wav_file_path + '.wav'
```
Comment (SevaSk) [inline TranscriberModels.py:37]: please rebase your branch, should not these changes unrelated to this feature in your PR

File: language.py [hunk 11]
```diff
@@ -0,0 +1,100 @@
+LANGUAGES ={"en":  "english",
```
Comment (SevaSk) [inline language.py:1]: 👍

**Review by SevaSk — CHANGES_REQUESTED** [review]
Added comment, sry cant spend time helping. I also don't have coding guidelines but try to follow best practices and feedback to get this approved.

File: AudioTranscriber.py (hunk 5 above)
Comment (SevaSk) [reply AudioTranscriber.py:97]: Sry meant to say that you added extra black line here unnecessary. Please remove this change

File: TranscriberModels.py [hunk 12]
```diff
@@ -11,17 +11,27 @@ def get_model(use_api):
...
     def get_transcription(self, wav_file_path):
         try:
-            result = self.audio_model.transcribe(wav_file_path, fp16=torch.cuda.is_available())
+            result = self.audio_model.transcribe(wav_file_path, fp16=torch.cuda.is_available(), language = self.lang)
         except Exception as e:
             print(e)
             return ''
         return result['text'].strip()
     
+    def change_lang(self, language):
```
Comment (SevaSk) [inline TranscriberModels.py:26]: Lets try to separate concerns here. This method does two things, it changes the language and return the audio model breaking single responsibility

**Review by SevaSk — CHANGES_REQUESTED** [review]
Also you should update the README to show that multiple language work with the local model only

File: TranscriberModels.py [hunk 13]
```diff
@@ -11,17 +11,33 @@ def get_model(use_api):
...
```
Comment (SevaSk) [inline TranscriberModels.py:26]:
```
def change_lang(self, language):
   self.lang = language
   load_model(language)
```

```
def load_model(language):
   if self.lang == "en":
       self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.en.pt'))
   else:
       self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.pt')) 
```

File: README.md [hunk 14]
```diff
@@ -93,7 +93,7 @@ While Ecoute provides real-time transcription and response suggestions, there ar
 
 **Whisper Model**: If the --api flag is not used, we utilize the 'tiny' version of the Whisper ASR model, due to its low resource consumption and fast response times. However, this model may not be as accurate as the larger models in transcribing certain types of speech, including accents or uncommon words.
 
-**Language**: The Whisper model used in Ecoute is set to English. As a result, it may not accurately transcribe non-English languages or dialects. We are actively working to add multi-language support to future versions of the program.
+**Language**: The Whisper model used in Ecoute is set to English. As a result, it may not accurately transcribe non-English languages or dialects. The multi-language support works with local models only.
```
Comment (SevaSk) [inline README.md:96]: Good to add that --api does not work with multiple languages above as well

File: TranscriberModels.py (hunk 13 above)
Comment (Adarsha-gg) [reply TranscriberModels.py:26]: I did know not to repeat myself but couldn't find this simple fix. Thank you!

File: TranscriberModels.py [hunk 15]
```diff
@@ -11,17 +11,30 @@ def get_model(use_api):
...
+        self.lang = language
+        self.load_model() 
+       
+    def load_model(self):
+         if self.lang == "en":
+            self.audio_model = whisper.load_model(os.path.join(os.getcwd(), 'tiny.en.pt'))
+            return self.audio_model
```
Comment (SevaSk) [inline TranscriberModels.py:33]: We dont need this return. Should just set the member variable here.

File: README.md [hunk 16]
```diff
@@ -93,7 +93,7 @@ While Ecoute provides real-time transcription and response suggestions, there ar
 
 **Whisper Model**: If the --api flag is not used, we utilize the 'tiny' version of the Whisper ASR model, due to its low resource consumption and fast response times. However, this model may not be as accurate as the larger models in transcribing certain types of speech, including accents or uncommon words.
 
-**Language**: If you are not using the --api flag the Whisper model used in Ecoute is set to English. As a result, it may not accurately transcribe non-English languages or dialects. We are actively working to add multi-language support to future versions of the program.
+**Language**: If you are not using the --api flag the Whisper model used in Ecoute is set to English. As a result, it may not accurately transcribe non-English languages or dialects. The multi-language support works with local models only and is not compatible with --api.
```
Comment (SevaSk) [inline README.md:96]: Sry for asking for this change! --api has multi-language support now so we can get rid of this limitation altogether.

File: main.py [hunk 17]
```diff
@@ -62,11 +63,14 @@ def create_ui_components(root):
...
-    return transcript_textbox, response_textbox, update_interval_slider, update_interval_slider_label, freeze_button
+    combobox = ctk.CTkOptionMenu(root, values=list(LANGUAGES.values()))                                  
+    combobox.grid(row=3, column=0, padx=200, pady=10, sticky="nsew")                                   
+                                        
+    return transcript_textbox, response_textbox, update_interval_slider, update_interval_slider_label, freeze_button, combobox
 
 def main():
     root = ctk.CTk()
-    transcript_textbox, response_textbox, update_interval_slider, update_interval_slider_label, freeze_button = create_ui_components(root)
+    transcript_textbox, response_textbox, update_interval_slider, update_interval_slider_label, freeze_button, combobox = create_ui_components(root)
```
Comment (SevaSk) [inline main.py:73]: Sry for asking for more changes incrementally but could this new box only appear when we are not in --api mode? Keep in mind good design practices and group this all --api and non --api functionally together.

**Review by SevaSk — CHANGES_REQUESTED** [review]

File: main.py [hunk 18]
```diff
@@ -79,6 +83,8 @@ def main():
     speaker_audio_recorder.record_into_queue(audio_queue)
 
     model = TranscriberModels.get_model('--api' in sys.argv)
+    combobox.configure(command=model.change_lang)
+
```
Comment (Truewarlock) [inline main.py:87]: Small tip: is this extra black line at 87 needed?

Comment (Adarsha-gg) [reply main.py:90]: No it isn't. Fixed it thanks!

File: main.py (hunk 17 above)
Comment (Adarsha-gg) [reply main.py:73]: I am new to this so I don't really know what good design practices are but I've tried to group them together.

**Review by Truewarlock — APPROVED** [review]

**Comment by Adarsha-gg** [comment]
I think that this is ready to be merged now @SevaSk

**Pull request closed without merging**